server/routes.go

```diff
@@ -720,9 +720,49 @@ func GetModelInfo(req api.ShowRequest) (*api.ShowResponse, error) {
 	fmt.Fprint(&sb, model.String())
 	resp.Modelfile = sb.String()
 
+	ggmlData, err := getGGMLData(model)
+	if err != nil {
+		return nil, err
+	}
+	resp.ModelInfo = ggmlData
+
 	return resp, nil
 }
 
+func getGGMLData(model *Model) ([]byte, error) {
+	f, err := os.Open(model.ModelPath)
+	if err != nil {
+		return nil, err
+	}
+
+	ggml, _, err := llm.DecodeGGML(f)
+	if err != nil {
+		return nil, err
+	}
+
+	f.Close()
```

Comment:
The best way to close a file is to defer right after open succeeds, e.g.

```go
f, err := os.Open(p)
if err != nil {
  return nil, err
}
defer f.Close()
```

This ensure the file is closed even when the function returns early, e.g. DecodeGGML errors